File: sdk/clojure/src/hitechcloud/sdk/virtualizor.clj
(ns hitechcloud.sdk.virtualizor
  (:require [hitechcloud.sdk.http-client :as http]))

;; Virtualizor: VPS management via Virtualizor panel

(defn suspend "POST /api/service/{id}/vms/{vmid}/suspend - Suspend virtual server" [client & [params]] (http-post client "/api/service/{id}/vms/{vmid}/suspend" params))

(defn unsuspend "POST /api/service/{id}/vms/{vmid}/unsuspend - Unsuspend virtual server" [client & [params]] (http-post client "/api/service/{id}/vms/{vmid}/unsuspend" params))

(defn list_rebuild_templates "GET /api/service/{id}/vms/{vmid}/rebuild - List rebuild templates" [client & [params]] (http-get client "/api/service/{id}/vms/{vmid}/rebuild" params))

(defn change_ssh_key "POST /api/service/{id}/vms/{vmid}/addsshkey - Change SSH key" [client & [params]] (http-post client "/api/service/{id}/vms/{vmid}/addsshkey" params))

File: sdk/fsharp/src/HiTechCloudSDK/Virtualizor.fs
namespace HiTechCloud.SDK

/// Virtualizor: VPS management via Virtualizor panel
type VirtualizorResource(httpClient: HttpClient) =

    /// POST /api/service/{id}/vms/{vmid}/suspend - Suspend virtual server
    member this.Suspend() =
        async {
            return! httpClient.postAsync("/api/service/{id}/vms/{vmid}/suspend")
        }

    /// POST /api/service/{id}/vms/{vmid}/unsuspend - Unsuspend virtual server
    member this.Unsuspend() =
        async {
            return! httpClient.postAsync("/api/service/{id}/vms/{vmid}/unsuspend")
        }

    /// GET /api/service/{id}/vms/{vmid}/rebuild - List rebuild templates
    member this.ListRebuildTemplates() =
        async {
            return! httpClient.getAsync("/api/service/{id}/vms/{vmid}/rebuild")
        }

    /// POST /api/service/{id}/vms/{vmid}/addsshkey - Change SSH key
    member this.ChangeSshKey() =
        async {
            return! httpClient.postAsync("/api/service/{id}/vms/{vmid}/addsshkey")
        }

